File: Lab5/Application/Services/DefaultRentalPolicy.cs
using Lab5.Domain.Entities;
using Lab5.Domain.Interfaces;

namespace Lab5.Application.Services;

public sealed class DefaultRentalPolicy : IRentalPolicy
{
    public DateOnly CalculateDueDate(LibraryUser user, DateOnly rentedOn)
    {
        var days = user switch
        {
            PremiumUser => 21,
            _ => 14
        };

        return rentedOn.AddDays(days);
    }
}

File: Lab5.Web/src/infrastructure/policies/DefaultRentalPolicy.js
export class DefaultRentalPolicy {
  calculateDueDate(user, rentedOn) {
    const days = user.maxActiveRentals > 3 ? 21 : 14;
    const dueDate = new Date(rentedOn);
    dueDate.setDate(dueDate.getDate() + days);
    return dueDate;
  }
}
